cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -272,21 +316,68 @@ type itemToPurge struct {
 	purge        func(skipPurge bool) error
 }
 
+func (p *BicepProvider) scopeForTemplate(ctx context.Context, t azure.ArmTemplate) (infra.Scope, error) {
+	switch t.TargetScope() {
+	case azure.DeploymentScopeSubscription:
+		return infra.NewSubscriptionScope(p.azCli, p.env.GetSubscriptionId()), nil
+	case azure.DeploymentScopeResourceGroup:
+		if !p.alphaFeatureManager.IsEnabled(ResourceGroupDeploymentFeature) {
+			return nil, ErrResourceGroupScopeNotSupported
+		}
+
+		if alpha.ShouldWarn(ResourceGroupDeploymentFeature) {
+			p.console.MessageUxItem(ctx, alpha.WarningMessage(ResourceGroupDeploymentFeature))
+		}
+
+		if p.env.Getenv(environment.ResourceGroupEnvVarName) == "" {
+			return nil, fmt.Errorf(
+				"%s must be set to the name of the resource group to use", environment.ResourceGroupEnvVarName,
+			)
+		}
+
+		return infra.NewResourceGroupScope(
+			p.azCli,
+			p.env.GetSubscriptionId(),
+			p.env.Getenv(environment.ResourceGroupEnvVarName),
+		), nil
+
+	default:
+		return nil, fmt.Errorf("unsupported deployment scope: %s", t.TargetScope())
+	}
+
+}
+
 // Destroys the specified deployment by deleting all azure resources, resource groups & deployments that are referenced.
 func (p *BicepProvider) Destroy(
 	ctx context.Context,
-	deployment *Deployment,
 	options DestroyOptions,
 ) *async.InteractiveTaskWithProgress[*DestroyResult, *DestroyProgress] {
 	return async.RunInteractiveTaskWithProgress(
 		func(asyncContext *async.InteractiveTaskContextWithProgress[*DestroyResult, *DestroyProgress]) {
+
+			modulePath := p.modulePath()
+			asyncContext.SetProgress(&DestroyProgress{Message: "Compiling Bicep template", Timestamp: time.Now()})
+			_, template, err := p.compileBicep(ctx, modulePath)
+			if err != nil {
+				asyncContext.SetError(fmt.Errorf("creating template: %w", err))
+				return
+			}
+
+			scope, err := p.scopeForTemplate(ctx, template)
+			if err != nil {
+				asyncContext.SetError(fmt.Errorf("computing deployment scope: %w", err))
+				return
+			}
+
 			asyncContext.SetProgress(&DestroyProgress{Message: "Fetching resource groups", Timestamp: time.Now()})
-			rgsFromDeployment, err := p.getResourceGroupsFromLatestDeployment(ctx)
+			deployment, err := latestCompletedDeployment(ctx, p.env.GetEnvName(), scope)
 			if err != nil {
 				asyncContext.SetError(err)
 				return
 			}
 
+			rgsFromDeployment := resourceGroupsFromDeployment(deployment)
```

Comment:
On line 416 below, we end up deleting the resource group. Would that be what we want in this case? Assuming that the deployer principal is scoped to a RG permission scope, I'm guessing the destroy will fail at that point?

Reply:
Yeah, it does indeed succeed (here's an example where I am destroying a resource group in our subscription, that I created to run the integration test I wrote in):

<img width="1565" alt="image" src="https://user-images.githubusercontent.com/9602953/236929285-462583d2-2904-4623-802b-9a9b8928e2be.png">

This seems like reasonable behavior to start with.